pkg/dockerregistry/server/blobdescriptorservice_test.go

```diff
@@ -0,0 +1,433 @@
+// context. If the object isn't given, its method will err out.
+func TestBlobDescriptorServiceIsApplied(t *testing.T) {
+	ctx := context.Background()
+
+	// don't do any authorization check
+	installFakeAccessController(t)
+	m := fakeBlobDescriptorService(t)
+	// to make other unit tests working
+	defer m.changeUnsetRepository(false)
+
+	testImage, err := registrytest.NewImageForManifest(t, "user/app", registrytest.SampleImageManifestSchema1, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	testImageStream := testNewImageStreamObject("user", "app", "latest", testImage.Name)
+	client := &testclient.Fake{}
+	client.AddReactor("get", "imagestreams", imagetest.GetFakeImageStreamGetHandler(t, *testImageStream))
+	client.AddReactor("get", "images", getFakeImageGetHandler(t, *testImage))
+
+	// TODO: get rid of those nasty global vars
+	backupRegistryClient := DefaultRegistryClient
+	DefaultRegistryClient = makeFakeRegistryClient(client, ktestclient.NewSimpleFake())
+	defer func() {
+		// set it back once this test finishes to make other unit tests working
+		DefaultRegistryClient = backupRegistryClient
+	}()
+
+	app := handlers.NewApp(ctx, &configuration.Configuration{
+		Loglevel: "debug",
+		Auth: map[string]configuration.Parameters{
+			fakeAuthorizerName: {"realm": fakeAuthorizerName},
+		},
+		Storage: configuration.Storage{
+			"inmemory": configuration.Parameters{},
+			"cache": configuration.Parameters{
+				"blobdescriptor": "inmemory",
+			},
+			"delete": configuration.Parameters{
+				"enabled": true,
+			},
+		},
+		Middleware: map[string][]configuration.Middleware{
+			"registry":   {{Name: "openshift"}},
+			"repository": {{Name: "openshift"}},
+			"storage":    {{Name: "openshift"}},
+		},
+	})
+	server := httptest.NewServer(app)
+	router := v2.Router()
+
+	serverURL, err := url.Parse(server.URL)
+	if err != nil {
+		t.Fatalf("error parsing server url: %v", err)
+	}
+	os.Setenv("DOCKER_REGISTRY_URL", serverURL.Host)
+
+	desc := registrytest.UploadTestBlob(t, serverURL, "user/app")
+
+	for _, tc := range []struct {
+		name                      string
+		method                    string
+		endpoint                  string
+		vars                      []string
+		unsetRepository           bool
+		expectedStatus            int
+		expectedMethodInvocations map[string]int
+	}{
+		{
+			name:     "get blob with repository unset",
+			method:   http.MethodGet,
+			endpoint: v2.RouteNameBlob,
+			vars: []string{
+				"name", "user/app",
+				"digest", desc.Digest.String(),
+			},
+			unsetRepository:           true,
+			expectedStatus:            http.StatusInternalServerError,
+			expectedMethodInvocations: map[string]int{"Stat": 1},
+		},
+
+		{
+			name:     "get blob",
+			method:   http.MethodGet,
+			endpoint: v2.RouteNameBlob,
+			vars: []string{
+				"name", "user/app",
+				"digest", desc.Digest.String(),
+			},
+			expectedStatus:            http.StatusOK,
+			expectedMethodInvocations: map[string]int{"Stat": 3},
```

Comment:
> how stable/valuable is this number? if it suddenly changes to 4 on a registry bump, will we investigate why or just change it everywhere?

It can be easily changed on the next bump. What I find valuable is a difference between number of calls with registry object set and unset. In the latter case it's always 1. In case of blobs it will be higher than 1. Shall I make the value an enumeration (`zero`, `one`, `morethanone`) instead?